library/books-frontend/src/components/BookCard.js

```diff
@@ -0,0 +1,40 @@
+import { Card, CardBody, CardExpandableContent, CardHeader, TextContent } from "@patternfly/react-core";
+import React, { Component } from "react";
+
+export default class BookCard extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      isExpanded: false,
+    };
+    this.onExpand = () => {
+      this.setState({
+        isExpanded: !this.state.isExpanded
+      });
+    };
+  }
+
+  render(){
+    const { year, copies, title } = this.props;
+    // const { authorName } = this.props;
+    const { isExpanded } = this.state;
+    return (
+      <>
+        <Card style={{margin: "20px"}} isExpanded={isExpanded} >
+          <CardHeader onExpand={this.onExpand}>
+            { title }
+          </CardHeader>
+          <CardExpandableContent>
+            <CardBody>
+              <TextContent>
+                {/* <p>Author: {authorName}</p> */}
+                <p>Year of publishing: {year}</p>
+                <p>Copies available: {copies}</p>
+              </TextContent>
+            </CardBody>
+          </CardExpandableContent>
+        </Card>
+      </>
+    );
+  }
+}
```

Comment:
With the addition of `useState`, I can't think of a reason to use class-based components over function-based components anymore 🙂 

```suggestion
export default function BookCard(props) {
  const { year, copies, title } = props;

  const [isExpanded, setIsExpanded] = useState(false);
  
  return (
    <>
      <Card style={{margin: "20px"}} isExpanded={isExpanded} >
        <CardHeader onExpand={() => setIsExpanded(!isExpanded)}>
          { title }
         </CardHeader>
         <CardExpandableContent>
           <CardBody>
             <TextContent>
               {/* <p>Author: {props.authorName}</p> */}
               <p>Year of publishing: {year}</p>
               <p>Copies available: {copies}</p>
             </TextContent>
           </CardBody>
         </CardExpandableContent>
       </Card>
       </>
  );
}
```

Reply:
+1, the original reason was for having multiple methods, but I cut them in the end, and you're right that the `onExpand` method was easy to bind by using an arrow function. Applying.